spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -99,9 +117,6 @@ protected void onInit() {
 	protected void doStart() {
 		Assert.state(this.outputChannel != null || StringUtils.hasText(this.outputChannelName),
 				"'outputChannel' or 'outputChannelName' is required");
```

Comment:
??? Do you need more explanations?
This is `MessageProducerSupport`, it can't be without an `outputChannel`.
So, move `assert` to the `getOutputChannel()` fully makes sense.
Since `GraphServer.afterSingletonsInstantiated()` is called already after all beans have processed (including the DSL parsing phase), we are safe there from the inconsistency of the configuration and from non-called `start()`.